File: python/leetcode/LEETCODE_14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        prefix: str = ""
        short: str = min(strs, key=lambda x: len(x))
        ind: int = 0
            
        while ind < len(short):
            
            for s in strs:
                if short[ind] != s[ind]:
                    return prefix
                
            prefix += short[ind]
            ind += 1
            
        return prefix

File: java/src/leetcode/LEETCODE_14.java
package leetcode;

import java.util.Arrays;
import java.util.Comparator;

class LEETCODE_14 {
    public String longestCommonPrefix(String[] strs) {

        String prefix = "";
        String shortest = Arrays.stream(strs)
            .min(Comparator.comparing(String::length))
            .get();
        
        int ind = 0;
        
        while (ind < shortest.length()) {
            
            for (String s : strs) {
                if (shortest.charAt(ind) != s.charAt(ind))
                    return prefix;
            }
            
            prefix += shortest.charAt(ind);
            ind++;
        }
        
        return prefix;
    }
}
